common/flagdproxy/flagdproxy.go

```diff
@@ -71,58 +76,99 @@ func (f *FlagdProxyHandler) Config() *FlagdProxyConfiguration {
+		}
+
+		// If the object is not found, we will create it
+		if notFound {
+			return f.Client.Create(ctx, obj)
+		}
+		// If the object exists but is not managed by OFO, return an error
+		if !common.IsManagedByOFO(old) {
+			return fmt.Errorf("%s not managed by OFO", obj.GetName())
+		}
+
+		// If the object is found, update if necessary
+		needsUpdate, err := specDiffers(obj, old)
+		if err != nil {
+			return err
+		}
+
+		if needsUpdate {
+			obj.SetResourceVersion(old.GetResourceVersion())
+			return f.Client.Update(ctx, obj)
+		}
+
+		return nil
+	})
 }
 
+// HandleFlagdProxy ensures flagd-proxy kubernetes components are configured properly
 func (f *FlagdProxyHandler) HandleFlagdProxy(ctx context.Context) error {
-	exists, deployment, err := f.doesFlagdProxyExist(ctx)
-	if err != nil {
-		return err
-	}
+	var err error
 
-	ownerReference, err := f.getOwnerReference(ctx)
+	ownerRef, err := f.getOwnerReference(ctx)
 	if err != nil {
 		return err
 	}
-	newDeployment := f.newFlagdProxyManifest(ownerReference)
-	newService := f.newFlagdProxyServiceManifest(ownerReference)
-
-	if !exists {
-		f.Log.Info("flagd-proxy Deployment does not exist, creating")
-		return f.deployFlagdProxy(ctx, f.createObject, newDeployment, newService)
-	}
-	// flagd-proxy exists, need to check if we should update it
-	if f.shouldUpdateFlagdProxy(deployment, newDeployment) {
-		f.Log.Info("flagd-proxy Deployment out of sync, updating")
-		return f.deployFlagdProxy(ctx, f.updateObject, newDeployment, newService)
-	}
-	f.Log.Info("flagd-proxy Deployment up-to-date")
-	return nil
-}
 
-func (f *FlagdProxyHandler) deployFlagdProxy(ctx context.Context, createUpdateFunc CreateUpdateFunc, deployment *appsV1.Deployment, service *corev1.Service) error {
-	f.Log.Info("deploying the flagd-proxy")
-	if err := createUpdateFunc(ctx, deployment); err != nil && !errors.IsAlreadyExists(err) {
+	if err = f.ensureFlagdProxyResource(ctx, f.newFlagdProxyDeployment(ownerRef)); err != nil {
 		return err
 	}
-	f.Log.Info("deploying the flagd-proxy service")
-	if err := createUpdateFunc(ctx, service); err != nil && !errors.IsAlreadyExists(err) {
+
+	if err = f.ensureFlagdProxyResource(ctx, f.newFlagdProxyService(ownerRef)); err != nil {
 		return err
 	}
-	return nil
+
+	err = f.ensureFlagdProxyResource(ctx, f.newFlagdProxyPodDisruptionBudget(ownerRef))
+	return err
 }
 
-func (f *FlagdProxyHandler) newFlagdProxyServiceManifest(ownerReference *metav1.OwnerReference) *corev1.Service {
+func (f *FlagdProxyHandler) newFlagdProxyService(ownerReference *metav1.OwnerReference) *corev1.Service {
 	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            FlagdProxyServiceName,
 			Namespace:       f.config.Namespace,
 			OwnerReferences: []metav1.OwnerReference{*ownerReference},
+			Labels: map[string]string{
+				common.ManagedByAnnotationKey: common.ManagedByAnnotationValue,
+			},
```

Comment:
I just noticed when upgrading to open-feature-operator that previous older flagdProxy resources that already existed did not have this label - and therefore ran in to the 'not managed by OFO' in the ensureFlagdProxyResource method. 

It was an easy workaround to manually add this label to the older resources but worth mentioning

Reply:
The previous resources did seem to have the `common.ManagedByAnnotationKey: common.ManagedByAnnotationValue,` in the Selector though